### PR TITLE
feature: support consumer bind plugins.

### PR DESCRIPTION
**Pull request by membphis**
- [x] test 

- [x] doc

### CODE REVIEW

File: doc/architecture-design-cn.md [hunk 1]
```diff
@@ -369,20 +369,72 @@ APISIX 区别于其他 API 网关的一大特点是允许用户选择不同 Rout
 
 如上图所示，作为 API 网关，需要知道 API Consumer（消费方）具体是谁，这样就可以对不同 API Consumer 配置不同规则。
 
+|字段|可选|说明|
+|---|----|----|
+|username|必选|Consumer 名称。|
```
Comment (juzhiyuan) [inline doc/architecture-design-cn.md]: 建议修改为 必选：是/否

Comment (membphis) [reply doc/architecture-design-cn.md]: good, fixed later

File: doc/architecture-design-cn.md [hunk 2]
```diff
@@ -369,20 +369,72 @@ APISIX 区别于其他 API 网关的一大特点是允许用户选择不同 Rout
...
 概括一下，Consumer 是某类服务的消费者，需与用户认证体系配合才能使用。
 比如不同的 Consumer 请求同一个 API，网关服务根据当前请求用户信息，对应不同的 Plugin 或 Upstream 配置。
 
-此外，大家也可以参考 [key-auth](doc/plugins/key-auth.md) 认证授权插件的调用逻辑，辅助大家来进一步理解 Consumer 概念和使用。
+此外，大家也可以参考 [key-auth](./plugins/key-auth.md) 认证授权插件的调用逻辑，辅助大家来进一步理解 Consumer 概念和使用。
+
+如何对某个 Consumer 开启指定插件，可以看下面例子：
+
+```shell
+# 创建 Consumer ，指定认证插件 key-auth ,并开启特定插件 limit-count
```
Comment (juzhiyuan) [inline doc/architecture-design-cn.md]: 应使用中文逗号

File: lua/apisix/plugin.lua [hunk 3]
```diff
@@ -280,57 +280,83 @@ function _M.stream_filter(user_route, plugins)
...
     end
 
-    return new_route_conf, true
+    core.log.info("merged conf : ", core.json.delay_encode(new_route_conf))
+    return new_route_conf or route_conf
+end
+
+
+function _M.merge_consumer_route(route_conf, consumer_conf)
+    core.log.info("route conf  : ", core.json.delay_encode(route_conf))
```
Comment (juzhiyuan) [inline lua/apisix/plugin.lua]: 建议统一 `log` 内容中的空格

Comment (moonming) [reply lua/apisix/plugin.lua]: @juzhiyuan good catch, and please use English, thx

**Pull request merged**